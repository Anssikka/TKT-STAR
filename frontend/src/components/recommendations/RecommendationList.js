import React from 'react'
import Book from '../books/Book'
import Blog from '../blogs/Blog'
import Video from '../videos/Video'
import '../../styles/RecommendationList.css'

function RecommendationList({books, videos, blogs, markAsRead }) {
  const mapToBooks = () => {
    if (books) {
      return books.map(book => (
        <Book key={book.id} handleToggle={() => markAsRead(book.id)} {...book} />
      ))
    }
    return []
  }

  const mapToVideos = () => {
    if (videos) {
      return videos.map(video => <Video key={video.id} {...video} />)
    }
    return []
  }

  const mapToBlogs = () => {
    if (blogs) {
      return blogs.map(blog => <Blog key={blog.id} {...blog}/>)
    }
    return []
  }

  return (
    <div className="recommendation-list">
      {mapToBooks()}
      {mapToVideos()}
      {mapToBlogs()}
    </div>
  )
}

export default RecommendationList
