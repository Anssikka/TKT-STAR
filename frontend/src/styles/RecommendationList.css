.recommendation-list {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.recommendation-list > * {
  width: 100vw;
  max-width: 35rem;
}
