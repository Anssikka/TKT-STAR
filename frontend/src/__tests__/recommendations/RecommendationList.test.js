import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup } from '@testing-library/react'
import RecommendationList from '../../components/recommendations/RecommendationList'
// import '@types/jest'

afterEach(cleanup)

describe('<RecommendationList/>', () => {
  it('has correct books and videos', () => {
    const books = [
      { id: 1, title: 'Book #1' },
      { id: 2, title: 'Book #2' }
    ]

    const videos = [
      { id: 1, title: 'Video #1' },
      { id: 2, title: 'Video #2' }
    ]
    const component = render(
      <RecommendationList books={books} videos={videos} markAsRead={jest.fn()} />
    )
    expect(component.asFragment()).toMatchSnapshot()
  })
})
