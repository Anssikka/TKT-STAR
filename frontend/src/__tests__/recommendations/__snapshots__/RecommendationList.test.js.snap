// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecommendationList/> has correct books and videos 1`] = `
<DocumentFragment>
  <div
    class="recommendation-list"
  >
    <div
      class="recommendation"
    >
      <h5>
        Book #1
      </h5>
      <button
        class="book-toggle toggle button null"
        data-testid="toggle"
      >
        Mark as read
      </button>
    </div>
    <div
      class="recommendation"
    >
      <h5>
        Book #2
      </h5>
      <button
        class="book-toggle toggle button null"
        data-testid="toggle"
      >
        Mark as read
      </button>
    </div>
    <div
      class="recommendation"
    >
      <h5>
        Video #1
      </h5>
    </div>
    <div
      class="recommendation"
    >
      <h5>
        Video #2
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
